### PR TITLE
pov: add test template

### LINKED ISSUE #1969: pov: add test template
## Task
Create a test template file for `pov`

### Details
1. If this is your first time contributing to this repository, welcome!
    - Contributing to a repository on GitHub requires a [fork](https://help.github.com/en/articles/fork-a-repo)
1. Create a branch on your local repository called `pov-add-test-template`
    - Run: `git checkout -b pov-add-test-template`
1. Clone problem-specifications repository; this is used to populate the template
    - Run: `git clone https://github.com/exercism/problem-specifications ../problem-specifications`
1. Create `exercises/pov/.meta/template.j2`
    - See [Generator Documentation] for more information
1. Generate test file from new template
    - Run: `bin/generate_tests.py pov`

At this point, if the generator fails, or you are otherwise stuck,
please feel free to reach out! It is perfectly acceptable to submit
an incomplete PR in order to get feedback or assistance.

### Resources
- [Contributing Guide]
- [Generator Documentation]
- [Closing issues using keywords]

[Contributing Guide]: https://github.com/exercism/python/blob/master/docs/CONTRIBUTING.md
[Generator Documentation]: https://github.com/exercism/python/blob/master/docs/GENERATOR.md
[Closing issues using keywords]: https://help.github.com/en/articles/closing-issues-using-keywords

### PR DESCRIPTION
**Pull request by crsmi**
Resolves #1969

### CODE REVIEW

File: exercises/pov/.meta/template.j2 [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        expected = {{ write_tree(case["expected"]) }}
+        self.assertTreeEquals(tree.{{ case["property"] | to_snake }}("{{ case["input"]["from"] }}"), expected)
+        {% else -%}
+        with self.assertRaisesWithMessage(ValueError):
+            tree.{{ case["property"] | to_snake }}("{{ case["input"]["from"] }}")
+        {% endif -%}
+{% endmacro -%}
+
+{%- macro test_path_to(case) -%}
+        {% if case["expected"] -%}
```
Comment (cmccandless) [inline exercises/pov/.meta/template.j2]:
```suggestion
        {% if not case is error_case -%}
```

File: exercises/pov/.meta/template.j2 [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        tree = {{ write_tree(case["input"]["tree"]) }}
+        {% if case["property"] == "fromPov" -%}
+        {{ test_from_pov(case) }}
+        {%- elif case["property"] == "pathTo" -%}
+        {{ test_path_to(case) }}
+        {%- endif -%}
+{% endmacro -%}
+
+{%- macro test_from_pov(case) -%}
+        {% if case["expected"] -%}
```
Comment (cmccandless) [inline exercises/pov/.meta/template.j2]:
```suggestion
        {% if not case is error_case -%}
```

File: exercises/pov/.meta/template.j2 [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        with self.assertRaisesWithMessage(ValueError):
+            tree.{{ case["property"] | to_snake }}("{{ case["input"]["from"] }}", "{{ case["input"]["to"]}}")
+        {% endif -%}
+{% endmacro -%}
+
+{%- macro write_tree(tree) -%}
+    Tree("{{ tree["label"] }}"
+    {%- if tree["children"] %}, [
+        {%- for child_tree in tree["children"] %}
+        {{ write_tree(child_tree) | indent(4,True,True) }}{% if not loop.last %}, {% endif %}
```
Comment (cmccandless) [inline exercises/pov/.meta/template.j2]: Good template code, but `black` will handle for formatting here, so might as well simplify the template.

```suggestion
        {{ write_tree(child_tree) | indent(4,True,True) }},
```

File: exercises/pov/.meta/template.j2 [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    {%- for case in supercase["cases"] %}
+    {{- test_case(case) }}
+    {%- endfor %}
+    {%- endfor %}
+
+# Custom Utility Functions
+    def assertTreeEquals(self, result, expected):
+          self.assertEqual(result, expected,
+                           '{} != {}'.format(result, expected))
+{{ macros.footer(True) }}
```
Comment (cmccandless) [inline exercises/pov/.meta/template.j2]: There's a built-in variable for this. Eventually, it won't need to be passed, but right now there's a bug in the global macros file.

```suggestion
{{ macros.footer(has_error_case) }}
```

File: exercises/pov/.meta/template.j2 (hunk 1 above)
Comment (crsmi) [reply exercises/pov/.meta/template.j2]: The canonical data for the pov exercise seems to indicate an error case by having the "expected" value be `null`. This means `error_case` will not work as expected. 

Should this be changed in the canonical data, or should we just work around it in the template and test file?

File: exercises/pov/.meta/template.j2 (hunk 4 above)
Comment (crsmi) [reply exercises/pov/.meta/template.j2]: Looks like `has_error_case` isn't working here for the same reason `error_case` won't work.

File: exercises/pov/.meta/template.j2 (hunk 1 above)
Comment (cmccandless) [reply exercises/pov/.meta/template.j2]: Hmm... this seems to have been a matter of interpretation by the original author of this exercise in the Python track. For now, keep your changes as they are, and we can revisit whether this behavior in the tests should be changed at a later date.

File: exercises/pov/.meta/template.j2 (hunk 4 above)
Comment (cmccandless) [reply exercises/pov/.meta/template.j2]: Agreed, keep explicit `True` argument.

File: exercises/pov/.meta/template.j2 (hunk 3 above)
Comment (cmccandless) [reply exercises/pov/.meta/template.j2]: @crsmi if we can just address this last comment, we should be good to merge.

**Pull request merged**